pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -583,6 +533,19 @@ func podSetUpdates(wl *kueue.Workload, pr *autoscaling.ProvisioningRequest) []ku
 	})
 }
 
+func (c *Controller) deactivateWorkload(ctx context.Context, wl *kueue.Workload, msg string) error {
+	updatedWl := &kueue.Workload{}
+	if err := c.client.Get(ctx, client.ObjectKeyFromObject(wl), updatedWl); err != nil {
+		return err
+	}
+	workload.SetEvictedCondition(updatedWl, kueue.WorkloadEvictedByDeactivation, msg)
+	if err := workload.ApplyAdmissionStatus(ctx, c.client, updatedWl, true); err != nil {
+		return err
+	}
+	updatedWl.Spec.Active = ptr.To(false)
+	return c.client.Update(ctx, updatedWl)
```

Comment:
I think this will cause conflicts in general. You need to get the workload again after doing an Apply.

Alternative, you can set `Active` to false via Patch.

Reply:
Fixed